physics/module_MYNNSFC_wrapper.F90

```diff
@@ -133,63 +112,69 @@ SUBROUTINE mynnsfc_wrapper_run(         &
+     &                      znt_ocn,   znt_lnd,   znt_ice,  &
+     &                      ust_ocn,   ust_lnd,   ust_ice,  &
+     &                       cm_ocn,    cm_lnd,    cm_ice,  &
+     &                       ch_ocn,    ch_lnd,    ch_ice,  &
+     &                       rb_ocn,    rb_lnd,    rb_ice,  &
+     &                   stress_ocn,stress_lnd,stress_ice,  &
+     &                       fm_ocn,    fm_lnd,    fm_ice,  &
+     &                       fh_ocn,    fh_lnd,    fh_ice,  &
+     &                     fm10_ocn,  fm10_lnd,  fm10_ice,  &
+     &                      fh2_ocn,   fh2_lnd,   fh2_ice,  &
+     &                     qsfc_ocn,  qsfc_lnd,  qsfc_ice
 
 !MYNN-2D
-      real(kind=kind_phys), dimension(im) ::                &
-     &        dx, pblh, slmsk, tsk, qsfc, ps,               &
-     &        zorl, ust, ustm, hflx, qfx, br, wspd, snowd,  &
+      real(kind=kind_phys), dimension(im), intent(in)    :: &
+     &        dx, pblh, slmsk, ps
+
+      real(kind=kind_phys), dimension(im), intent(inout) :: &
+     &        ustm, hflx, qflx, wspd, qsfc,                 &
      &        FLHC, FLQC, U10, V10, TH2, T2, Q2,            &
      &        CHS2, CQS2, rmol, zol, mol, ch,               &
-     &        fm, fh, fm10, fh2,                            &
-     &        lh, cda, cka, stress, wstar
+     &        lh, wstar
      !LOCAL
       real, dimension(im) ::                                &
-     &        qcg, hfx, znt, ts, snowh, psim, psih,         &
-     &        chs, ck, cd, mavail, regime, xland, GZ1OZ0       
+     &        hfx, znt, ts, psim, psih,                     &
+     &        chs, ck, cd, mavail, xland, GZ1OZ0,           &
+     &        cpm, qgh, qfx
 
       ! Initialize CCPP error handling variables
       errmsg = ''
       errflg = 0
 
-      if (lprnt) then
-         write(0,*)"=============================================="
-         write(0,*)"in mynn surface layer wrapper..."
-         write(0,*)"flag_init=",flag_init
-         write(0,*)"flag_restart=",flag_restart
-         write(0,*)"iter=",iter
-      endif
-
-      ! If initialization is needed and mynnsfc_wrapper is called
-      ! in a subcycling loop, then test for (flag_init==.T. .and. iter==1);
-      ! initialization in sfclay_mynn is triggered by itimestep == 1
-      ! DH* TODO: Use flag_restart to distinguish which fields need
-      ! to be initialized and which are read from restart files
-      if (flag_init.and.iter==1) then
-          itimestep = 1
-      else
-          itimestep = 2
-      endif
+!      if (lprnt) then
+!         write(0,*)"=============================================="
+!         write(0,*)"in mynn surface layer wrapper..."
+!         write(0,*)"flag_init=",flag_init
+!         write(0,*)"flag_restart=",flag_restart
+!         write(0,*)"iter=",iter
+!      endif
 
       !prep MYNN-only variables
-            do k=1,levs
+            do k=1,2 !levs
```

Comment:
Is this really intended (running up to 2 only instead of levs)?

Reply:
yep, we only need to initialize values at k= 1 and 2.

Reply:
Great, thanks.